File: tool/measurementDriver/src/ch/ethz/ruediste/roofline/dom/WorkloadStopEventPredicate.java
package ch.ethz.ruediste.roofline.dom;

public class WorkloadStopEventPredicate extends WorkloadStopEventPredicateData {
	private Workload workload;

	public WorkloadStopEventPredicate() {
		super();
	}

	public WorkloadStopEventPredicate(Workload workload) {
		this();
		this.workload = workload;
	}

	@Override
	public int getWorkloadId() {
		return workload.getId();
	}

	public Workload getWorkload() {
		return workload;
	}

	public void setWorkload(Workload workload) {
		this.workload = workload;
	}
}

File: tool/measuringCore/src/sharedEntities/eventPredicates/WorkloadStopEventPredicate.cpp
/*
 * WorkloadStopRule.cpp
 *
 *  Created on: Feb 9, 2012
 *      Author: ruedi
 */

#include "WorkloadStopEventPredicate.h"
#include "baseClasses/WorkloadStopEvent.h"
#include "sharedEntities/Workload.h"

#include <typeinfo>

bool WorkloadStopEventPredicate::doesMatch(EventBase *event)
{
	// check if it is a stop event
	if (typeid(*event)==typeid(WorkloadStopEvent)){
		WorkloadStopEvent *stopEvent=(WorkloadStopEvent*)event;

		// check the workload id
		if (getWorkload()==NULL || stopEvent->getWorkloadId()==getWorkload()->getId()){
			return true;
		}
	}
	return false;
}

